### PR TITLE
Add double and Unit tests

### PR DESCRIPTION
**Pull request by baiyu123**

### CODE REVIEW

File: KidtronicsLanguage/var.cpp [hunk 1]
```diff
@@ -10,13 +10,13 @@
 #include <string.h>
 #include <stdlib.h>
 #include "var.h"
+#include <iostream>
```
Comment (baixiaobest) [inline KidtronicsLanguage/var.cpp]: I don't think you can compile this library with arduino.

File: KidtronicsLanguage/var.cpp [hunk 2]
```diff
@@ -54,87 +59,100 @@ var& var::operator=(const var& other) {
     }
     return *this;
 }
+/** Get value in double so we can compare int with double. */
+double var::getValueInDouble() const {
```
Comment (baixiaobest) [inline KidtronicsLanguage/var.cpp]: You can rename it to getValueAsDouble.

File: KidtronicsLanguage/var.cpp [hunk 3]
```diff
@@ -182,19 +184,43 @@ var operator+(var lhs, const var& rhs) {
...
+        else{
+            return lhs;
+        }
     }
-    else {
+    else if(lhs.m_currentType == var::VarType::DOUBLE){
+        if(rhs.m_currentType == var::VarType::INTEGER || rhs.m_currentType == var::VarType::DOUBLE){
+            return var(lhs.getValueInDouble() - rhs.getValueInDouble());
+        }
+        else{
```
Comment (baixiaobest) [inline KidtronicsLanguage/var.cpp]: You can remove this else block and the other following else block. Then have the function return lhs when program reaches the end of the function.

File: KidtronicsLanguage/var.cpp [hunk 4]
```diff
@@ -182,19 +184,43 @@ var operator+(var lhs, const var& rhs) {
...
-    {
-        return var(lhs.m_integerValue - rhs.m_integerValue);
+    if (lhs.m_currentType == var::VarType::INTEGER) {
+        if(rhs.m_currentType == var::VarType::INTEGER){
+            return var(lhs.m_integerValue - rhs.m_integerValue);
+        }
+        else if(rhs.m_currentType == var::VarType::DOUBLE){
+            return var(lhs.getValueInDouble() - rhs.getValueInDouble());
+        }
+        else{
```
Comment (baixiaobest) [inline KidtronicsLanguage/var.cpp]: This else block can be removed.

File: KidtronicsLanguage/var.cpp [hunk 5]
```diff
@@ -182,19 +184,43 @@ var operator+(var lhs, const var& rhs) {
...
-    else {
+    else if(lhs.m_currentType == var::VarType::DOUBLE){
+        if(rhs.m_currentType == var::VarType::INTEGER || rhs.m_currentType == var::VarType::DOUBLE){
+            return var(lhs.getValueInDouble() - rhs.getValueInDouble());
+        }
+        else{
+            return lhs;
+        }
+    }
+    else{
```
Comment (baixiaobest) [inline KidtronicsLanguage/var.cpp]: Remove this else and have the function return lhs

File: KidtronicsLanguage/var.h [hunk 6]
```diff
@@ -43,6 +53,10 @@ class var {
     
     // Assignment operator.
     var& operator=(const var& other);
+    // Get value in double
+    double getValueInDouble() const;
```
Comment (baixiaobest) [inline KidtronicsLanguage/var.h]: Make this function private, user of the class will never need this.

**Review by baixiaobest — CHANGES_REQUESTED** [review]
Very good, still need some changes.

File: KidtronicsLanguage/var.cpp (hunk 1 above)
Comment (baiyu123) [reply KidtronicsLanguage/var.cpp]: It was for debug, I thought I removed it, but it show up

File: KidtronicsLanguage/var.cpp (hunk 2 above)
Comment (baiyu123) [reply KidtronicsLanguage/var.cpp]: ok

File: KidtronicsLanguage/var.h (hunk 6 above)
Comment (baiyu123) [reply KidtronicsLanguage/var.h]: It will be used in other class, for example, left hand side try to compare itself with right hand side.

**Pull request merged**